core-overlay.html

```diff
@@ -111,666 +124,373 @@
+      // animation
+      renderOpened: function() {
+        // this.notifyResize();
+        this.renderBackdropOpened();
+        // var transition = this.getTransition();
+        // if (transition) {
+        //   transition.go(this, {opened: this.opened});
+        // } else {
+          this.transitionend();
+        // }
+      },
+
+      // finishing tasks; typically called via a transition
+      transitionend: function(e) {
+        // make sure this is our transition event.
+        if (e && e.target !== this) {
+          return;
+        }
+        this.transitioning = false;
+        if (!this.opened) {
+          this.resetTargetDimensions();
+          this.style.display = 'none';
+          this.completeBackdrop();
+          manager.removeOverlay(this);
         }
-        trackBackdrop(this);
-      }
-    },
 
-    renderBackdropOpened: function() {
-      if (this.backdrop && getBackdrops().length < 2) {
-        this.scrim.classList.toggle('core-opened', this.opened);
-      }
-    },
+        this.applyFocus();
 
-    completeBackdrop: function() {
-      if (this.backdrop) {
-        trackBackdrop(this);
-        if (getBackdrops().length === 0) {
-          this.scrim.parentNode.removeChild(this.scrim);
-        }
-      }
-    },
+        // complete all tasks before firing -completed event
+        this.fire('core-overlay-' + (this.opened ? 'open' : 'close') +
+            '-completed');
+      },
 
-    preparePositioning: function() {
-      this.target.style.transition = this.target.style.webkitTransition = 'none';
-      this.target.style.transform = this.target.style.webkitTransform = 'none';
-      this.target.style.display = '';
-    },
+      prepareBackdrop: function() {
+        if (this.backdrop && this.opened) {
+          if (!this._backdrop.parentNode) {
+            Polymer.dom(document.body).appendChild(this._backdrop);
+            this._backdrop.style.zIndex = manager.currentOverlayZ() - 1;
+          }
+          manager.trackBackdrop(this);
+        }
+      },
 
-    discoverDimensions: function() {
-      if (this.dimensions) {
-        return;
-      }
-      var target = getComputedStyle(this.target);
-      var sizer = getComputedStyle(this.sizingTarget);
-      this.dimensions = {
-        position: {
-          v: target.top !== 'auto' ? 'top' : (target.bottom !== 'auto' ?
-            'bottom' : null),
-          h: target.left !== 'auto' ? 'left' : (target.right !== 'auto' ?
-            'right' : null),
-          css: target.position
-        },
-        size: {
-          v: sizer.maxHeight !== 'none',
-          h: sizer.maxWidth !== 'none'
-        },
-        margin: {
-          top: parseInt(target.marginTop) || 0,
-          right: parseInt(target.marginRight) || 0,
-          bottom: parseInt(target.marginBottom) || 0,
-          left: parseInt(target.marginLeft) || 0
+      renderBackdropOpened: function() {
+        if (this.backdrop && manager.getBackdrops().length < 2) {
```

Comment:
This conditional seems a little magic-number-y at a glance. Can you clarify why `1` or `0` backdrops is a special case?